src/server/core/launch_server.c

```diff
@@ -50,6 +51,7 @@ int launch_server(data_t *data)
         return 84;
     printf("Port : %i\n", data->port);
     listen_to_inputs(&my_addr, server_fd, data);
+    detect_win(data, 1);
```

Comment:
This function will not be called if a player wins because it's outside the run loop (line 27 of this file).
Please move it between lines 33 and 34.